src/Dependencies/Collections/Internal/ICollectionCalls.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+
+namespace Microsoft.CodeAnalysis.Collections.Internal
+{
+    internal static class ICollectionCalls
+    {
+        public static bool IsSynchronized<TCollection>(ref TCollection collection)
```

Comment:
What is this? How does it connect with the rest of ImmutableSegmentList?

Reply:
⏱️ It invokes `ICollection.IsSynchronized` on a value type that explicitly implements this property without requiring the instance be copied or boxed. I'll add a comment.